core/block_svg.js

```diff
@@ -1025,7 +1025,9 @@ Blockly.BlockSvg.prototype.handleDragFree_ = function(oldXY, newXY, e) {
   }
 
   var updatePreviews = true;
-  if (Blockly.localConnection_ && Blockly.highlightedConnection_) {
+  if (localConnection && localConnection.type == Blockly.OUTPUT_VALUE) {
+    updatePreviews = true; // Always update previews for output connections.
+  } else if (Blockly.localConnection_ && Blockly.highlightedConnection_) {
```

Comment:
...and in turn, this should probably be 

var updatePreviews = true;
if (!(localConnection && localConnection.type == Blockly.OUTPUT_VALUE) &&
(Blockly.localConnection_ && Blockly.highlightedConnection_)) {

since the first clause is a no-op.

If you want to leave it this way for clarity, that's fine too.